src/runtime_src/core/tools/xbmgmt2/SubCmdProgram.cpp

```diff
@@ -730,6 +733,33 @@ SubCmdProgram::execute(const SubCmdOptions& _options) const
     throw xrt_core::error("uuid does not match BLP");
   }
 
+  // -- process "user" option ---------------------------------------
+  if(!xclbin.empty()) {
+    XBU::verbose(boost::str(boost::format("  xclbin: %s") % xclbin));
+    //only 1 card and name
+    if(deviceCollection.size() > 1)
+      throw xrt_core::error("Please specify a single device");
+    auto dev = deviceCollection.front();
+
+    std::ifstream stream(xclbin, std::ios::binary);
+    if (!stream)
+      throw xrt_core::error(boost::str(boost::format("Could not open %s for reading") % xclbin));
+
+    stream.seekg(0,stream.end);
+    ssize_t size = stream.tellg();
+    stream.seekg(0,stream.beg);
+
+    std::vector<char> xclbin_buffer(size);
+    stream.read(xclbin_buffer.data(), size);
```

Comment:
Please use xrt::xclbin to read the xclbin.
Let's rename xrt_core::load_xclbin() to reflect different loading different from the 'normal' xclbin loading.